source/dem/particle_particle_heat_transfer.cc

```diff
@@ -0,0 +1,236 @@
+                                   const double effective_youngs_modulus,
+                                   const double effective_real_youngs_modulus,
+                                   const double normal_force_norm)
+{
+  const double contact_radius = pow((3 * normal_force_norm * effective_radius) /
+                                      (4 * effective_youngs_modulus),
+                                    (1.0 / 3.0));
+
+  // apply correction
+  return contact_radius *
+         pow(effective_youngs_modulus / effective_real_youngs_modulus,
+             1.0 / 5.0);
+}
+
+double
+calculate_macrocontact_resistance(const double harmonic_particle_conductivity,
+                                  const double contact_radius)
+{
+  return 1 / (2.0 * contact_radius * harmonic_particle_conductivity + DBL_MIN);
+}
+
+double
+calculate_microcontact_resistance(const double equivalent_surface_slope,
+                                  const double equivalent_surface_roughness,
+                                  const double effective_microhardness,
+                                  const double contact_radius,
+                                  const double harmonic_particle_conductivity,
+                                  const double maximum_pressure)
+{
+  return 1.184 /
+         (M_PI * harmonic_particle_conductivity * contact_radius *
+          contact_radius) *
+         (equivalent_surface_roughness / equivalent_surface_slope) *
+         pow(effective_microhardness / (maximum_pressure + DBL_MIN), 0.96);
+}
+
+double
+calculate_solid_macrogap_resistance(const double radius_one,
+                                    const double radius_two,
+                                    const double thermal_conductivity_one,
+                                    const double thermal_conductivity_two,
+                                    const double contact_radius)
+{
+  // resistance_i = characteristic length parallel to heat flux /
+  // (thermal_conductivity
+  // * characteristic area perpendicular to the heat flux)
+  const double resistance_one =
+    (M_PI * radius_one / 4.0) * 1 /
+    (M_PI * (radius_one * radius_one - contact_radius * contact_radius) *
+     thermal_conductivity_one);
+  const double resistance_two =
+    (M_PI * radius_two / 4.0) * 1 /
+    (M_PI * (radius_two * radius_two - contact_radius * contact_radius) *
+     thermal_conductivity_two);
+
+  return resistance_one + resistance_two;
+}
+
+double
+erfc_inverse_approximation(const double x)
+{
+  if (x < 1e-9 || x > 1.9)
+    {
+      std::cerr
+        << "Error. Input out of range (10^-9 <= x <= 1.9) for erfc-1 function in thermal resistance calculation."
+        << std::endl;
+      return NAN;
+    }
+
+  if (x <= 0.02)
+    {
+      return 1.0 / (0.218 + 0.735 * pow(x, 0.173));
+    }
+  else if (x <= 0.5)
+    {
+      return (1.05 * pow(0.175, x)) / pow(x, 0.12);
+    }
+  else if (x <= 1.9)
+    {
+      return (1 - x) / (0.707 + 0.862 * x - 0.431 * x * x);
+    }
```

Comment:
A sequence of conditions like that (that will be evaluated extremely frequently), ideally, if you know what is solved and the general order of magnitude of the terms, ensures that the first condition is the most likely one to be true.

Reply:
In this case since it's a < , < , < condition this current order is fine, you can then let the branch predictor do the trick :)